Updates cell styles

ryanlanciaux opened the pull request:
This PR updates the Cell styles. In this paradigm, the Cell actions are no longer in a floating toolbar and are now in a popout menu. 

![cell](https://user-images.githubusercontent.com/85041/84173825-cd308680-aa4b-11ea-84ca-2a826a5630a3.gif)

## TODO
There are a couple of things that need to occur before moving this from a Draft to a final PR.

- [ ] Update tests
- [x] Run the application in more scenarios
- [x] Ensure that the Toolbar does not expand outside the visible area when on the last Cell.



<!-- If this is your first PR for nteract, please mark these boxes to confirm (otherwise you can exclude them) -->

<!-- Thank you for submitting a pull request to nteract. After all, open source is powered by contributors like you! -->

<!-- Before you submit your PR, make sure that you have gone through the following checklist to ensure that everything goes smoothly. -->

<!-- If you're PR is not fully polished yet or you'd like to park some work, you can open a draft PR. -->

- [x] I have read the [Contributor Guide](https://github.com/nteract/nteract/blob/master/CONTRIBUTING.md)
- [ ] I have updated the changelogs/current_changelog.md file with some information about the change that I am making the appropriate file.
- [ ] I have validated or unit-tested the changes that I have made.
- [ ] I have run through the TEST_PLAN.md to ensure that my change does not break anything else.

<!-- Questions? Feel free to ping us on https://nteract.slack.com/ -->

captainsafia commented:
Gave this a spin. Exciting to see the new changes.

One piece of feedback I have is the interaction with the input is hidden.

Currently, it just hides the text. Typically, users will want to entire input area to be hidden and the toolbar elements to be moved adjacent to the output.

Also, I feel like there should be a bit of a margin in the output so that it lines up with the cell toolbar.

ryanlanciaux commented:
@captainsafia Thanks a ton for this feedback! Do you think it would make sense if I switched the layout to where the toolbar elements is on the left of both the input and the output? I think that may cover both the margin and condensing things a bit when the input is hidden. 

Thanks!

captainsafia commented:
>  Do you think it would make sense if I switched the layout to where the toolbar elements is on the left of both the input and the output?

I think I am visualizing the same thing you are with this description and it sounds like a sensible approach.

ryanlanciaux commented:
Okay great! I've updated this PR with the styling updates so the sidebar stays next to the output when the input is hidden:

![image](https://user-images.githubusercontent.com/85041/84654161-b0c59b80-aedc-11ea-935f-0d105ea2bbd1.png)

Additionally, I've updated the color of the sidebar. I'm still seeing some failing tests, and I'm trying to determine how this may have broken them.

captainsafia commented on packages/presentational-components/package.json:
```diff
@@ -9,7 +9,9 @@
     "@blueprintjs/select": "^3.2.0",
-    "react-syntax-highlighter": "^12.0.0"
+    "react-syntax-highlighter": "^12.0.0",
+    "@types/classnames": "^2.2.10",
```
Let's move this to `devDependencies`.

captainsafia commented on packages/presentational-components/package.json:
```diff
@@ -26,9 +28,11 @@
     "css": "^2.2.4",
-    "styled-components": "^4.4.1"
+    "styled-components": "^4.4.1",
+    "react-toggle": "^4.1.1"
```
This should be under `dependencies` I believe.

captainsafia commented on packages/notebook-app-component/src/derived-components/toolbar.tsx:
```diff
@@ -5,151 +5,340 @@
+                  <Icons.Play />
+                </CircularButton>
               ) : null}
-              <DropdownMenu
```
We might want to set some tabindexes on the menu items.

I can tab from the `Play` button to the `Toolbar` icon and press enter to show the toolbar. From that point on, I cannot tab through the menu options in the toolbar.

ryanlanciaux replied:
I think this should be working now.

captainsafia commented on packages/stateful-components/src/cells/toolbar.tsx:
```diff
@@ -43,11 +45,9 @@ class CellToolbar extends React.Component<
 > {
   render() {
     return (
-      <div className="nteract-cell-toolbar">
```
The `nteract-cell-toolbar` classname is part of the public API that we provide as part of stateful components so we will need to keep it around.

Was there a particular reason it was removed?

ryanlanciaux replied:
Sorry about that - there was no real reason. I've added it back in.

captainsafia commented on packages/styles/themes/default.css:
```diff
@@ -56,19 +56,51 @@
 }
 
-.nteract-cell {
+.nteract-cell{
```
```suggestion
.nteract-cell {

```

captainsafia commented on packages/styles/themes/default.css:
```diff
@@ -56,19 +56,51 @@
+  font-family: 'Source Code Pro', sans-serif;
+}
+
+.nteract-cell-row{
```
```suggestion
.nteract-cell-row {

```

captainsafia commented on packages/styles/themes/default.css:
```diff
@@ -56,19 +56,51 @@
+  width: 100%;
+}
+
+.nteract-cell-container:focus .nteract-cell, .nteract-cell-container:focus-within .nteract-cell, .nteract-cell-container.selected .nteract-cell{
```
```suggestion
.nteract-cell-container:focus .nteract-cell,
.nteract-cell-container:focus-within .nteract-cell,
.nteract-cell-container.selected .nteract-cell {
```

captainsafia commented on packages/styles/themes/default.css:
```diff
@@ -56,19 +56,51 @@
   color: var(--theme-app-fg);
```
Nit: Can you keep this file unchanged and move the styles related to a new theme to a new file?

ryanlanciaux replied:
That would definitely make things a bit cleaner. I've created a CSS file just to contain these styles.

captainsafia commented:
Merging this now so that we can tackle to follow-ons in smaller PRs.
